### PR TITLE
[feat] Add ability to adjust the density of the visible content in tables

### PR DESCRIPTION
**Pull request by roubkar**
Following changes provide the ability to control the content density in Explorer tables

- Change row height sizes
- Rename row height modes names
- Decrease font size for compact mode
- Decrease `Badge` height for `xSmall` size

### CODE REVIEW

**Comment by roubkar** [comment]
Resolves https://github.com/aimhubio/aim/issues/1741

File: aim/web/ui/src/components/kit/DataList/DataList.scss [hunk 1]
```diff
@@ -55,9 +55,9 @@
 
       &-cell {
         position: relative;
-        min-height: 32px;
+        min-height: 28px;
         max-height: 300px;
-        padding: toRem(7px) toRem(16px);
+        padding: 0 toRem(16px);
```
Comment (arsengit) [inline aim/web/ui/src/components/kit/DataList/DataList.scss:60]: lets use @space variables

File: aim/web/ui/src/config/enums/tableEnums.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 enum RowHeightEnum {
-  small = 28,
-  medium = 32,
-  large = 36,
+  small = 22,
```
Comment (arsengit) [inline aim/web/ui/src/config/enums/tableEnums.ts:2]: What about changing the small height to 21, like in google sheets

**Review by arsengit — CHANGES_REQUESTED** [review]
On normal height mode

Screen Shot 2022-05-13 at 21 14 37

File: aim/web/ui/src/config/enums/tableEnums.ts (hunk 2 above)
Comment (roubkar) [reply aim/web/ui/src/config/enums/tableEnums.ts:2]: 21 is odd number it doesn't go well with other tokens of our system

**Review by VkoHov — APPROVED** [review]
lgtm!

**Comment by YodaEmbedding** [comment]
For more compactness, I personally either hide the badges or make them really small:

```css
@-moz-document regexp("^https?://(127.0.0.1|localhost)(:\d*)?(/?|/.*)$") {
  .Table__cell--header {
    padding: .375rem 0.2rem;
  }

  .Table__cell--header > :first-child {
    margin-right: 0rem;
    font-size: 0.6rem;  /* Default: 0.875rem */
  }

  .Table__cell--header .Badge__label {
    max-width: 3em;
  }

  .Table__cell__value.isNumeric {
    justify-content: flex-start;
    max-width: 3em;
  }

  /* 3 vertical dots button */
  .Table__cell--header .MuiButton-root {
    width: 1.0rem !important;
    min-width: 1.0rem !important;
  }
}
```

**Comment by roubkar** [comment]
Hey @YodaEmbedding, thanks for sharing this.

We also think that Aim UI components can be more compact and currently are in the process of revisiting our design-system tokens (especially sizes and spaces)

**Pull request merged**